website/client/components/chat/chatMessages.vue

```diff
@@ -135,6 +137,9 @@ export default {
       if (!message.flagCount || message.flagCount < 2) return true;
       return this.user.contributor.admin;
     },
+    canViewUnapprovedChat (message) {
+      return !this.approvalRequired(message) || this.user.contributor.admin;
+    },
```

Comment:
I'm checking with the other mods to see if we also want to let the user see their own message. That wasn't specified in the issue but I've realised from my testing that it's confusing when you make a post and don't see it appear in the chat.

Reply:
We've decided that the user should be able to see their own post before it's approved, and should see a message on it indicating that it's pending approval. That will avoid confusion about why their post doesn't appear / why no one is replying and will hopefully prevent them posting the same message over and over again to get attention. Details are here in the issue: https://github.com/HabitRPG/habitica/issues/8063#issuecomment-385624004